packages/react-core/src/scalprum-component.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { Fragment, useEffect, Suspense, useState } from 'react';
+import { getApp, getAppsByRootLocation, injectScript } from '@scalprum/core';
+import { loadComponent } from './async-loader';
+
+export interface ScalprumComponentProps<T = Record<string, unknown>> {
+  fallback?: string;
+  appName: string;
+  path: string;
+  api?: T;
+  scope: string;
+  module: string;
+}
+
+export const ScalprumComponent: React.ComponentType<ScalprumComponentProps> = ({
+  fallback = 'loading',
+  appName,
+  path,
+  api,
+  scope,
+  module,
+  ...props
+}) => {
+  const { scriptLocation } = getAppsByRootLocation(path as string)?.[0];
+  const [Component, setComponent] = useState<React.ComponentType<any>>(Fragment);
+  const [mountedAt, setMountedAt] = useState<HTMLScriptElement | undefined>();
+  useEffect(() => {
+    const app = getApp(appName);
+
+    if (!app) {
+      injectScript(appName, scriptLocation).then(([, scriptMountedAt]) => {
+        const app = getApp(appName);
+        app?.mount<JSX.Element>(api);
+        setComponent(() => React.lazy(loadComponent(scope, module)));
+        setMountedAt(() => scriptMountedAt);
+      });
+    } else {
+      app?.mount<JSX.Element>(api);
+      setComponent(() => React.lazy(loadComponent(scope, module)));
+    }
+    return () => {
+      const app = getApp(appName);
+      app?.unmount();
+      mountedAt && document.body.removeChild(mountedAt);
```

Comment:
I think this will no trigger React component lifecycle and can leave the instance in the virtual DOM. Can you check it in browser?

Reply:
I's on unmount, so when component unmounts it removes the script that initiated it.